File: src/test/java/gameoflife/app/GameOfLifeUITest.java
package gameoflife.app;

import gameoflife.helper.IOHelperKt;

import java.util.logging.Logger;

class GameOfLifeUITest extends GameOfLifeUI {
    private static final int ITERATIONS = 500;
    private static final Logger LOG = Logger.getLogger(GameOfLifeUITest.class.getName());

    private int iterations = ITERATIONS;

    private GameOfLifeUITest(String[] params) {
        super(params);
    }

    @Override
    public boolean isContinueFlag() {
        return iterations-- >= 0;
    }

    public static void main(String[] params) {
        final long time = System.currentTimeMillis();
        new GameOfLifeUITest(params).setWaitTime(0).run();
        LOG.info("Total time: " + IOHelperKt.format(System.currentTimeMillis() - time));
    }
}

File: src/test/kotlin/gameoflife/app/GameOfLifeUITest.kt
package gameoflife.app

import gameoflife.helper.*

import java.util.logging.Logger
import kotlin.system.measureTimeMillis

internal class GameOfLifeUITest private constructor(params: Array<String>) : ContinueCheck {

    private var iterations = ITERATIONS
    private val gameOfLifeUI: GameOfLifeUI = GameOfLifeUI(params)

    init {
        gameOfLifeUI.setWaitTime(50)
        gameOfLifeUI.isContinue = this
    }

    override fun isContinue(): Boolean {
        return iterations-- >= 0
    }

    companion object {
        private const val ITERATIONS = 500
        private val LOG = Logger.getLogger(GameOfLifeUITest::class.java.name)

        @JvmStatic
        suspend fun main(params: Array<String>) {
            measureTimeMillis { GameOfLifeUITest(params).gameOfLifeUI.run() }
                .apply { LOG.info("Total time: $toTime") }
        }
    }
}
